testsuite/cluster-test/src/experiments/client_compatibility_test.rs

```diff
@@ -5,65 +5,151 @@
 
 use crate::{
     cluster::Cluster,
+    cluster_swarm::cluster_swarm_kube::CFG_SEED,
     experiments::{Context, Experiment, ExperimentParam},
+    instance::Instance,
 };
 use async_trait::async_trait;
 use libra_logger::prelude::*;
-use std::{collections::HashSet, fmt, time::Duration};
+use libra_types::chain_id::ChainId;
+use std::{collections::HashSet, fmt, iter::once, time::Duration};
 use structopt::StructOpt;
 use tokio::time;
 
 #[derive(StructOpt, Debug)]
-pub struct ClientCompatiblityTestParams {}
+pub struct ClientCompatiblityTestParams {
+    #[structopt(long, help = "Image tag of old client to test")]
+    pub old_image_tag: String,
+}
 
-pub struct ClientCompatibilityTest {}
+pub struct ClientCompatibilityTest {
+    old_image_tag: String,
+    test_host: Instance,
+    test_node: Instance,
+}
 
 impl ExperimentParam for ClientCompatiblityTestParams {
     type E = ClientCompatibilityTest;
-    fn build(self, _cluster: &Cluster) -> Self::E {
-        Self::E {}
+    fn build(self, cluster: &Cluster) -> Self::E {
+        let (test_nodes, _) = cluster.split_n_fullnodes_random(2);
```

Comment:
will it be good to not hard code size = 2 here?

Reply:
My plan was to gradually increase this number. e.g. to test on more nodes, and also to test other clients like pylibra.